File: solutions/0538. Convert BST to Greater Tree/0538.cpp
class Solution {
 public:
  TreeNode* convertBST(TreeNode* root) {
    int prefixSum = 0;

    reversedInorder(root, prefixSum);

    return root;
  }

 private:
  void reversedInorder(TreeNode* root, int& prefixSum) {
    if (!root)
      return;

    reversedInorder(root->right, prefixSum);
    prefixSum += root->val;
    root->val = prefixSum;
    reversedInorder(root->left, prefixSum);
  }
};

File: solutions/0538. Convert BST to Greater Tree/0538.java
class Solution {
  public TreeNode convertBST(TreeNode root) {
    reversedInorder(root);
    return root;
  }

  private int prefixSum = 0;

  private void reversedInorder(TreeNode root) {
    if (root == null)
      return;

    reversedInorder(root.right);
    prefixSum += root.val;
    root.val = prefixSum;
    reversedInorder(root.left);
  }
}